packages/fast-form-generator-react/src/form/form.utilities.ts

```diff
@@ -218,6 +227,160 @@ export function getComponentTracker(
+    normalizedDataLocation = convertArrayItemsToBracketNotation(dataLocation, data); // convert back all array items to use [ ]
+    const dataLocationStrings: string[] = normalizedDataLocation.split(".");
+
+    if (dataLocation !== "") {
+        dataLocationStrings.unshift("");
+    }
+
+    const schemaLocationStrings: string[] = getSchemaLocationStringsFromDataLocationStrings(dataLocationStrings, schema, data);
+
+    return schemaLocationStrings.join(".").replace(squareBracketRegex, "");
+}
+
+/**
+ * Get an array of schema location strings from an array of data location strings
+ */
+export function getSchemaLocationStringsFromDataLocationStrings(dataLocationStrings: string[], schema: any, data: any): string[] {
+    const squareBracketRegex: RegExp = /\[(\d+?)\]/g;
+    let schemaLocationStrings: string[] = [];
+    let reconstitutedDataLocation: string = "";
+
+    for (let i: number = 0; i < dataLocationStrings.length; i++) {
+        const partialData: any = getPartialData(reconstitutedDataLocation, data);
+        const partialSchema: any = getPartialData(schemaLocationStrings.join(".").replace(squareBracketRegex, ""), schema);
+
+        schemaLocationStrings = schemaLocationStrings.concat(
+            getSchemaOneOfAnyOfLocationStrings(
+                partialSchema,
+                partialData
+            )
+        );
+
+        if (dataLocationStrings[i] !== "") {
+            schemaLocationStrings = schemaLocationStrings.concat(
+                getSchemaLocationStringsFromDataLocationString({
+                    isObject: typeof partialData === "object",
+                    isArray: Array.isArray(partialData),
+                    dataLocation: dataLocationStrings[i],
+                    data: partialData,
+                    endOfContainingString: dataLocationStrings.length > i + 1
+                })
+            );
+        }
+
+        reconstitutedDataLocation += reconstitutedDataLocation === "" ? dataLocationStrings[i] : `.${dataLocationStrings[i]}`;
+    }
+
+    return schemaLocationStrings;
+}
+
+/**
+ * Gets data from a data and location
+ */
+export function getPartialData(location: string, data: any): any {
+    return location === "" ? data : get(data, location);
+}
+
+/**
+ * Gets an array of oneOf/anyOf with a valid index from a schema and data
+ */
+export function getSchemaOneOfAnyOfLocationStrings(schema: any, data: any): string[] {
+    const schemaLocationStrings: string[] = [];
+
+    if (!!schema.anyOf) {
+        schemaLocationStrings.push(`anyOf.${getValidAnyOfOneOfIndex("anyOf", data, schema)}`);
+    }
+
+    if (!!schema.oneOf) {
+        schemaLocationStrings.push(`oneOf.${getValidAnyOfOneOfIndex("oneOf", data, schema)}`);
+    }
+
+    return schemaLocationStrings;
+}
+
+/**
+ * Get an array of schema location strings from a single data location item
+ */
+export function getSchemaLocationStringsFromDataLocationString(config: ISchemaLocationStringsFromDataLocationStringConfig): string[] {
```

Comment:
`getSchemaLocationStringsFromDataLocationString` doesn't seem like an accurate name - I'd expect it to accept a string as an argument but it doesn't